implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -232,6 +249,84 @@ private String influxLineProtocol(Meter.Id id, String metricType, Stream<Field>
                 + " " + clock.wallTime();
     }
 
+    private void recognizeInfluxDBVersion() {
+
+        if (influxDBVersion != null) {
+            return;
+        }
+
+        String uri = config.uri() + "/ping";
+        try {
+            httpClient
+                    .head(uri)
+                    .send()
+                    .onSuccess(response -> {
+                        logger.debug("InfluxDB successfully pinged: '{}'", response.headers());
+                        influxDBVersion = response.headers().entrySet()
+                                .stream()
+                                .filter(entry -> "X-Influxdb-Version".equalsIgnoreCase(entry.getKey()))
+                                .map(entry -> {
+                                    boolean v1 = entry.getValue().stream().anyMatch(value -> value.startsWith("1."));
+                                    return v1 ? InfluxDBVersion.V1 : InfluxDBVersion.V2;
+                                })
+                                .findFirst()
+                                // There is no header "X-Influxdb-Version: 1.X" => use v2 API
+                                .orElse(InfluxDBVersion.V2);
+                        logger.debug("InfluxDB version configured to: '{}'", influxDBVersion);
+                    })
+                    .onError(response -> {
+                        logger.error("unable to ping InfluxDB: '{}'. Use v2 API.: {}", uri, response.body());
+                        // Unable ping http://server/ping => use v2 API
+                        influxDBVersion = InfluxDBVersion.V2;
+                    });
+        } catch (Throwable e) {
+            logger.error("unable to ping InfluxDB: '{}', Use v1 API.", uri, e);
```

Comment:
What exception are we anticipating that would end up in this catch block? We are defaulting to v2 above if there is an error calling the ping endpoint, so why is this defaulting to v1?

Reply:
> What exception are we anticipating that would end up in this catch block? 

Defaulting version to `v1`.

> We are defaulting to v2 above if there is an error calling the ping endpoint, so why is this defaulting to v1?

This implementation was prepared for early version of InfluxDB v2 which didn't supports the `/ping` endpoint. For the current version it doesn't sense => I changed the implementation. Now the the default version is set to `v1`.

Reply:
A problem with only determining the version from the results of a ping endpoint is that if the ping request happens to fail on the first publish, then the version will be set to v1 for the application, even if you wanted v2 and configured things for v2 and the influxdb backend is available after the ping request.

Since we aren't trying to retrieve the org anymore, perhaps we should add a config for `apiVersion` and have it default to v2 if `org` is configured, v1 otherwise. This way users can explicitly set the influx API version they intend to target.

Reply:
Changed. The determining of InfluxDB version is based on the config.